packages/braintree-utils/src/utils/get-valid-braintree-fastlane-styles.ts

```diff
@@ -0,0 +1,142 @@
+import { BraintreeFastlaneStylesOption, FastlaneStylesSettings } from '../index';
+
+export default function getValidBraintreeFastlaneStyles(
+    styles?: FastlaneStylesSettings,
+): BraintreeFastlaneStylesOption | undefined {
+    if (!styles) {
+        return undefined;
+    }
+
+    const fastlaneStyles: BraintreeFastlaneStylesOption = {};
+
+    getValidRootSettings(styles, fastlaneStyles);
+    getValidInputSettings(styles, fastlaneStyles);
+    getValidToggleSettings(styles, fastlaneStyles);
+    getValidTextStyles(styles, fastlaneStyles);
+
+    return Object.keys(fastlaneStyles).length ? fastlaneStyles : {};
```

Comment:
We should avoid making such side effects and keep function as pure as possible. I recommend to return an object from each function that you use and place the result into fastlaneStyles object.

```suggestion
    if (!styles) {
        return undefined;
    }

    const rootStyles = getValidRootSettings(styles);
    const ... = getValidInputSettings(styles);
    const ... = getValidToggleSettings(styles);
    const ... = getValidTextStyles(styles);

    return {
        ...(rootStyles || {}),
        and so on
    };
```